packages/framework/src/Model/Product/Search/Export/ProductSearchExportConsumer.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Product\Search\Export;
+
+use Doctrine\ORM\EntityManagerInterface;
+use OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface;
+use OldSound\RabbitMqBundle\RabbitMq\DequeuerInterface;
+use PhpAmqpLib\Message\AMQPMessage;
+
+class ProductSearchExportConsumer implements ProductSearchExportConsumerInterface, ConsumerInterface
+{
+    /**
+     * @var \OldSound\RabbitMqBundle\RabbitMq\DequeuerInterface
+     */
+    protected $dequeuer;
+
+    /**
+     * @var \Doctrine\ORM\EntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportFacade
+     */
+    protected $productSearchExportFacade;
+
+    /**
+     * @param \OldSound\RabbitMqBundle\RabbitMq\DequeuerInterface $dequeuer
+     * @param \Doctrine\ORM\EntityManagerInterface $entityManager
+     * @param \Shopsys\FrameworkBundle\Model\Product\Search\Export\ProductSearchExportFacade $productSearchExportFacade
+     */
+    public function __construct(
+        DequeuerInterface $dequeuer,
+        EntityManagerInterface $entityManager,
+        ProductSearchExportFacade $productSearchExportFacade
+    ) {
+        $this->dequeuer = $dequeuer;
+        $this->entityManager = $entityManager;
+        $this->productSearchExportFacade = $productSearchExportFacade;
+    }
+
+    /**
+     * @param \PhpAmqpLib\Message\AMQPMessage $message
+     * @return int
+     */
+    public function execute(AMQPMessage $message)
+    {
+        $productId = (int)$message->getBody();
+
+        $this->entityManager->clear();
+
+        $this->productSearchExportFacade->exportIds([$productId]);
```

Comment:
It is as you say. There are possible performance consequences but they should not be significant. I cannot tell you exact consequences because we have no performance tests for this right now.